Support for Cloud Storage

faizananwerali opened the pull request:
Support for Cloud Storage (Google Bucket/Amazon S3) without breaking local and public storage

Fixes https://github.com/the-control-group/voyager/issues/4477

faizananwerali committed:
Support for Cloud Storage

Support for Cloud Storage (Google Bucket/Amazon S3) without breaking local and public storage

faizananwerali committed:
Merge pull request #1 from faizananwerali/faizananwerali-patch-1

Support for Cloud Storage

faizananwerali committed:
Update VoyagerMediaController.php

faizananwerali committed:
Update VoyagerMediaController.php

faizananwerali committed:
Update VoyagerMediaController.php

faizananwerali committed:
Update VoyagerMediaController.php

fletch3555 commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -302,13 +307,16 @@ function ($constraint) use ($thumbnail_data) {
                                 $thumbnail = $this->addWatermarkToImage($thumbnail, $details->watermark);
                             }
                             $thumbnail->save($realPath.$request->upload_path.$name.'-'.($thumbnail_data->name ?? 'thumbnail').'.'.$extension, ($details->quality ?? 90));
+                            //$path = $realPath.$request->upload_path.$name.'-'.($thumbnail_data->name ?? 'thumbnail').'.'.$extension;
```
Please remove these comments.

faizananwerali replied:
fixed in latest commit

fletch3555 commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -302,13 +307,16 @@ function ($constraint) use ($thumbnail_data) {
                         $image = $this->addWatermarkToImage($image, $details->watermark);
                     }
                     $image->save($realPath.$file, ($details->quality ?? 90));
+                    //Storage::disk($this->filesystem)->put($realPath.$file, (string) $image->encode(null, ($details->quality ?? 90)));
```
Same as above

faizananwerali replied:
fixed in latest commit

fletch3555 commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -339,34 +347,69 @@ public function crop(Request $request)
     }
 
-    private function addWatermarkToImage($image, $options)
+    private function addWatermarkToImage(\Intervention\Image\Image $image, $options)
```
`Image` is already imported.  No need to use the full namespace here.

faizananwerali replied:
fixed in latest commit

fletch3555 commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -339,34 +347,69 @@ public function crop(Request $request)
+
+        if (!$success) {
+            try {
+                $content = Storage::disk($this->filesystem)->get($options->source);
```
This is the same as line 398 above.  Isn't it guaranteed to fail again?  This whole block seems unnecessary.

faizananwerali replied:
fixed in latest commit

MrCrayon commented:
Honestly I think this is the wrong approach, upload works so we know we can get the right path that works in all storages without all trial and error
What I would like to see is full path for what works:

- upload
- delete?
- move?
- rename?

And full path for what doesn't work, then we can spot the difference and fix it.

faizananwerali commented:
Can I give you amazon s3 and Google Bucket Iamuser credentials? Because I
literally don't have anytime this month to debug and both on local and live
server

MrCrayon commented:
@faizananwerali you can contact me in Voyager Slack Channel

faizananwerali committed:
Fix Media Manager for Cloud storage

Fixed by MrCrayon. https://github.com/the-control-group/voyager/compare/1.3...MrCrayon:03f2af479612e20718cacfcf4c5240b2e982049b
Image can't read from cloud so we need to get open the image with Storage and pass the content to Image, inverted process for saving.
Storage should be used without $realPath.
$image->encode()->encoded returns the image as string.

faizananwerali committed:
Fix Media Manager for Cloud storage

Fixed by MrCrayon. the-control-group/voyager@1.3...MrCrayon:03f2af479612e20718cacfcf4c5240b2e982049b
Image can't read from cloud so we need to get open the image with Storage and pass the content to Image, inverted process for saving.
Storage should be used without $realPath.
$image->encode()->encoded returns the image as string.

faizananwerali committed:
Fix Media Manager for Cloud storage

Fixed by MrCrayon. the-control-group/voyager@1.3...MrCrayon:03f2af479612e20718cacfcf4c5240b2e982049b
Image can't read from cloud so we need to get open the image with Storage and pass the content to Image, inverted process for saving.
Storage should be used without $realPath.
$image->encode()->encoded returns the image as string.

faizananwerali committed:
Support for Cloud Storage

Fixed by MrCrayon. the-control-group/voyager@1.3...MrCrayon:03f2af479612e20718cacfcf4c5240b2e982049b
Image can't read from cloud so we need to get open the image with Storage and pass the content to Image, inverted process for saving.
Storage should be used without $realPath.
$image->encode()->encoded returns the image as string.

faizananwerali committed:
Fix Media Manager for Cloud storage by MrCrayon

Fixed by MrCrayon. the-control-group/voyager@1.3...MrCrayon:03f2af479612e20718cacfcf4c5240b2e982049b
Image can't read from cloud so we need to get open the image with Storage and pass the content to Image, inverted process for saving.
Storage should be used without $realPath.
$image->encode()->encoded returns the image as string.

faizananwerali commented:
Fixed by MrCrayon. https://github.com/MrCrayon/voyager/commit/03f2af479612e20718cacfcf4c5240b2e982049b
Image can't read from cloud so we need to get open the image with Storage and pass the content to Image, inverted process for saving.
Storage should be used without $realPath.
$image->encode()->encoded returns the image as string.

faizananwerali committed:
Merge branch '1.3' into 1.3

faizananwerali committed:
Merge branch '1.3' into 1.3

MrCrayon commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -255,8 +255,8 @@ public function upload(Request $request)
             ];
             if (in_array($request->file->getMimeType(), $imageMimeTypes)) {
-                $image = Image::make($realPath.$file);
-
```
Can you restore the blank lines as the original?
It makes diff much more readable, you can compare with this: https://github.com/MrCrayon/voyager/commit/1a55e5653bfb3d09ce99ef812070bb35fe616d11

MrCrayon commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -338,22 +339,23 @@ public function crop(Request $request)
                 // override the original image
                 $destImagePath = $originImagePath;
             }
-
```
Same as above

MrCrayon commented on src/Http/Controllers/VoyagerMediaController.php:
```diff
@@ -246,7 +246,7 @@ public function upload(Request $request)
             }
 
             $file = $request->file->storeAs($request->upload_path, $name.'.'.$extension, $this->filesystem);
-
```
Same as above

MrCrayon reviewed (changes requested):
Please fix the empty lines for clarity.

faizananwerali committed:
Restored extra lines

Restored extra lines as request https://github.com/the-control-group/voyager/pull/4565#pullrequestreview-335267497

faizananwerali committed:
Restored extra lines

Restored extra lines as requested by MrCrayon https://github.com/the-control-group/voyager/pull/4565#pullrequestreview-335267497

faizananwerali committed:
Merge branch '1.3' into 1.3

MrCrayon reviewed (dismissed):
Thanks :+1:

faizananwerali committed:
Merge branch '1.3' into 1.3

faizananwerali commented:
I don't understand the holdup for merging. Can you please merge it master branch so I don't have to override class

faizananwerali commented:
what the heck do I have to do to get this pull request approved? I mean I don't understand what do you want more.

emptynick committed:
Merge branch '1.4' into 1.3

emptynick commented:
Thanks. But please, watch your tone.
We are all volunteers and have jobs.
It sometimes just takes some time.
